src/client/pfs.go

```diff
@@ -762,6 +762,46 @@ func (c APIClient) Compact() error {
 	return err
 }
 
+type GetFileClient interface {
+	GetFile(repoName, commitID, path string, offset, size int64, writer io.Writer) error
+
+	//TODO(kdelga): Figure out if we need these.
+	//GetFileReader(repoName string, commitID string, path string, offset int64, size int64) (io.Reader, error)
+
+	//GetFileReadSeeker(repoName string, commitID string, path string) (io.ReadSeeker, error)
+}
+
+type getFileClient struct {
+	c pfs.API_GetFileStreamClient
+	mu sync.Mutex
+	oneoff bool
+}
+
+func (c APIClient) newOneoffGetFileClient(repoName string, commitID string, path string, offset int64, size int64 /*, writer io.Writer*/) (GetFileClient, error) {
+	//gfc, err := c.PfsAPIClient.GetFileStream(c.Ctx())
+	gfc, err := c.getFileStream(repoName, commitID, path, offset, size)
+
+	if err != nil {
+		return nil, grpcutil.ScrubGRPC(err)
+	}
+	return &getFileClient{c: gfc, oneoff: true}, nil
+}
+
+//TODO(kdelga): It smells bad to pass the receiver as an arg (see other interface funcs too)
+func (c getFileClient) GetFile(repoName, commitID, path string, offset, size int64, writer io.Writer) error {
+	if err := grpcutil.WriteFromStreamingBytesClient(c, writer); err != nil {
+		return grpcutil.ScrubGRPC(err)
+	}
+	return nil
+}
+
+func (c getFileClient) Recv() (*types.BytesValue, error) {
+	gfr, err := c.c.Recv()
```

Comment:
Am I on the right track implementing `Recv()` in the client and `Send()` in the server (see below)? It seems a little fishy that they would be implemented in different places and I don't see any precedent for this elsewhere, but it works as is.

Reply:
That seems a little weird but not necessarily wrong. Is there a reason `Send` can't be implemented here? I'm assuming that `Recv` basically needs to be implemented so that this type has the right signature to be used by the `grpcutil` functions?

Reply:
Nope, I don't think there is any reason why Send can't be implemented. I think I originally did it because it seemed kind of weird to have something called GetFilesServer implemented in the client and then imported and used by the server, but I agree it makes more sense to put it here because all our wrappers around protobuf generated types live in the client package.

You are correct that these exist in order to be used by grpcutil.